src/Nethermind/Nethermind.JsonRpc/Data/TransactionForRpc.cs

```diff
@@ -38,7 +38,14 @@ public TransactionForRpc(Keccak? blockHash, long? blockNumber, int? txIndex, Tra
         }
         ChainId = transaction.ChainId;
         Type = transaction.Type;
-        AccessList = transaction.AccessList is null ? null : AccessListItemForRpc.FromAccessList(transaction.AccessList);
+        if (transaction.SupportsAccessList)
+        {
+            AccessList = transaction.AccessList is null ? Array.Empty<AccessListItemForRpc>() : AccessListItemForRpc.FromAccessList(transaction.AccessList);
+        }
+        else
+        {
+            AccessList = null;
+        }
```

Comment:
Main change for this PR: when a Tx supports `accessList` then `AccessList` should be allways non-null; otherwise the field should be always null.